### ISSUE TITLE
Cannot read property 'push' of undefined (import/no-named-as-default)

### ISSUE THREAD

**Issue by chris-flanagan** [benmosher/eslint-plugin-import#1089]
I'm using eslint-config-airbnb-base along with eslint-plugin-import.

./test
```js
class Test {}

export default Test;
```

./index
```js
import Test from './test';
                  ^^^^^^
```

Parse errors in imported module './test': Cannot read property 'push' of undefined (undefined:undefined) (import/no-named-as-default)
Parse errors in imported module './test': Cannot read property 'push' of undefined (undefined:undefined) (import/no-named-as-default-member)

I've read every answer on the web, I don't understand what way it's looking for?

**Comment by sross0988** [comment]
I got this error myself and fixed it by upgrading to `"babel-eslint": "^10.0.2",`

What happens is the error is thrown from `./node_modules/babel-eslint/lib/patch-eslint-scope.js`

In the line `estraverse.VisitorKeys.MethodDefinition.push("decorators");`, `estraverse.VisitorKeys.MethodDefinition` does not exist so it tries to push to undefined.

Looking at the `babel-eslint` release log they identified and fixed this issue in `10.0.0` and above where they say _"Drop monkeypatching behavior"_.
See:
https://github.com/babel/babel-eslint/releases/tag/v10.0.0
https://github.com/babel/babel-eslint/pull/689/files

**Comment by ljharb** [comment]
Sounds like it's not an error in the import plugin, then. Thanks!
